op-supervisor/supervisor/backend/db/db.go

```
@@ -147,11 +159,31 @@ func NewChainsDB(l log.Logger, depSet depset.DependencySet, m Metrics) *ChainsDB
 		m = metrics.NoopMetrics
 	}
 
-	return &ChainsDB{
-		logger: l,
-		depSet: depSet,
-		m:      m,
+	db := &ChainsDB{
+		logger:          l,
+		depSet:          depSet,
+		m:               m,
+		activationCheck: activation.NewCheck(depSet, l),
+	}
+
+	// Pre-initialize chains in pre-activation mode if appropriate
+	for _, chainID := range depSet.Chains() {
+		canActivate, err := depSet.CanInitiateAt(chainID, uint64(time.Now().Unix()))
+		if err != nil {
+			l.Warn("Failed to check activation status for chain", "chain", chainID, "err", err)
+			continue
+		}
```

Does it make sense to continue here if the check fails? Shouldn't we return the error instead?